pkg/policy/k8s/service.go

```diff
@@ -311,12 +296,38 @@ func (s *serviceEndpoints) processRule(rule *api.Rule) (numMatches int) {
 		for _, toService := range egress.ToServices {
 			if sel := toService.K8sServiceSelector; sel != nil {
 				if serviceSelectorMatches(sel, s.svcID, s.svc) {
-					appendEndpoints(&rule.Egress[i].ToCIDRSet, s.endpoints())
+					if s.svc.IsExternal() {
+						appendEndpoints(&rule.Egress[i].ToCIDRSet, s.endpoints())
+					} else {
+						svcSelector := make(map[string]string)
+						for k, v := range s.svc.Selector {
+							svcSelector[k] = v
+						}
+						if sel.Namespace != "" {
```

Comment:
It possible that the namespace is set to an empty string, and that will match services in all namespaces with the name given - but not all endpoints matching the service selector clusterwide.

So I think here you should use `s.svc.Namespace` instead - and we know a service is always namespaced, so that should never be an empty string.

Reply:
Done.